### PR TITLE
feat: cloud notification config & notification type IDs

### PR DESCRIPTION
**Pull request by chris13524**
Introduces a new `id` field for notification types which allows `name` to be human-readable.

`id` must contain only lowercase letters, numbers, hyphens, or underscores. This allows our existing set of notifications (`gm_hourly`), as well as auto-incrementing number and uuid IDs.

Context: https://walletconnect.slack.com/archives/C044SKFKELR/p1696459959133769

### CODE REVIEW

**Review by Cali93 — APPROVED** [review]

**Review by devceline — COMMENTED** [review]
Maybe we should keep `name` optional? If it doesn't exist we use `id` for display purposes

**Review by Elyniss — APPROVED** [review]
LGTM

**Review by flypaper0 — APPROVED** [review]

**Comment by chris13524** [comment]
Closing in-favor of moving this config to cloud.

**Comment by chris13524** [comment]
Reopened to add cloud config method to specs, and including other tweaks to improve the spec

File: docs/specs/clients/notify/notify-config.md [hunk 1]
```diff
@@ -1,14 +1,25 @@
 # Notify Configuration
 
-In order to enable notify, dapps must expose a static json file named `wc-notify-config.json` in their `.well-known` path under their domain URL.
+## Cloud configuration
 
-The file should include the following information:
+```
+GET https://explorer-api.walletconnect.com/w3i/v1/notify-config?projectId=<project-id>&app_domain=<app-domain>
```
Comment (Cali93) [inline docs/specs/clients/notify/notify-config.md]:
```suggestion
GET https://explorer-api.walletconnect.com/w3i/v1/notify-config?projectId=<project-id>&appDomain=<app-domain>
```

File: docs/specs/clients/notify/notify-config.md [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+```
+
+- projectId - project ID of the project requesting the config
+- app_domain - domain of the notify app the config is being requested for (e.g. app.example.com)
```
Comment (Cali93) [inline docs/specs/clients/notify/notify-config.md]:
```suggestion
- appDomain - domain of the notify app the config is being requested for (e.g. app.example.com)
```

**Review by Cali93 — APPROVED** [review]

**Pull request merged**